sys/include/json.h

```diff
@@ -0,0 +1,725 @@
+ *          Calling json_write_array_next() directly after opening the array
+ *          is valid but not needed.
+ *          Calling json_write_array_next() before closing the array with
+ *          json_write_array_close() without putting an item is invalid.
+ * @see json_write_object_open()
+ * @see json_write_array_next()
+ * @param[in] cookie The cookie that was initialized with json_write_init().
+ * @returns @ref JSON_OKAY on success.
+ * @returns @ref JSON_INVALID_DATA if the stream is inside an invalid state.
+ * @returns @ref JSON_PREMATURELY_ENDED if the data could not be written.
+ */
+json_result_t json_write_array_close(json_write_cookie_t *cookie);
+
+/**
+ * @brief Close an array.
+ * @details See description in json_write_array_close().
+ * @see json_write_object_open()
+ * @see json_write_array_close()
+ * @param[in] cookie The cookie that was initialized with json_write_init().
+ * @returns @ref JSON_OKAY on success.
+ * @returns @ref JSON_INVALID_DATA if the stream is inside an invalid state.
+ * @returns @ref JSON_PREMATURELY_ENDED if the data could not be written.
+ */
+json_result_t json_write_array_next(json_write_cookie_t *cookie);
+
+/**
+ * @brief Write a single integer value.
+ * @details JSON does not differentiate between integers and floating point numbers.
+ * @warning JSON is an unframed format. If the data to deliver consists of a single number,
+ *          then there is a fair chance that it will not be processed correctly by the decoder side.
+ * @see json_write_float()
+ * @see json_write_int64()
+ * @param[in] cookie The cookie that was initialized with json_write_init().
+ * @param[in] value The value to put into the data stream.
+ * @returns @ref JSON_OKAY on success.
+ * @returns @ref JSON_INVALID_DATA if the stream is inside an invalid state.
+ * @returns @ref JSON_PREMATURELY_ENDED if the data could not be written.
+ */
+json_result_t json_write_int(json_write_cookie_t *cookie, int32_t value);
+
+/**
+ * @brief Write a floating point number.
+ * @details JSON does not differentiate between integers and floating point numbers.
+ * @warning The number must be finite, i.e. it must not be NaN, +Inf or -Inf.
+ *          Illegal inputs are not caught but the stream will be broken.
+ * @warning JSON is an unframed format. If the data to deliver consists of a single number,
+ *          then there is a fair chance that it will not be processed correctly by the decoder side.
+ * @see json_write_int()
+ * @see json_write_int64()
+ * @param[in] cookie The cookie that was initialized with json_write_init().
+ * @param[in] value The value to put into the data stream.
+ * @returns @ref JSON_OKAY on success.
+ * @returns @ref JSON_INVALID_DATA if the stream is inside an invalid state.
+ * @returns @ref JSON_PREMATURELY_ENDED if the data could not be written.
+ */
+json_result_t json_write_float(json_write_cookie_t *cookie, float value);
+
+#ifndef MODULE_ATMEGA_COMMON
```

Comment:
Why is this hardware dependent?

Reply:
ATMEGA did not support int64_t values at all. (In 2014, maybe my toolchain was to blame for that.)

Reply:
will check and adapt as needed

Reply:
In any case I would prefer a NOP or error message instead of excluding this function completely from the API.